kore/test/Test/Kore/Step/Axiom/Matcher.hs

```diff
@@ -1029,6 +1043,813 @@ test_unificationWithAppMatchOnTop =
+                              )
+                            ]
+                        }
+                    ]
+        actual <- matchCons
+            (mkVar Mock.xMap `cons` mkMap [])
+            [ (1, 11)
+            , (2, 12)
+            , (3, 13)
+            ]
+        assertEqualWithExplanation "" expect actual
+    , testCase "cons (var, unit) vs concrete" $ do
+        let expect =
+                Just $ MultiOr.make
+                    [ Conditional
+                        { term = ()
+                        , predicate = makeTruePredicate
+                        , substitution = Substitution.unsafeWrap
+                            [ (Mock.xMap, mkMap
+                                  [ (mkKey 1, mkVal 11)
+                                  , (mkKey 3, mkVal 13)
+                                  ]
+                              )
+                            ]
+                        }
+                    ]
+        actual <- matchCons
+            (mkVar Mock.xMap `cons` mkMap [(mkKey 2, mkVal 12)])
+            [ (1, 11)
+            , (2, 12)
+            , (3, 13)
+            ]
+        assertEqualWithExplanation "" expect actual
+    , testCase "cons (var, concrete) vs concrete" $ do
+        let expect =
+                Just $ MultiOr.make
+                    [ Conditional
+                        { term = ()
+                        , predicate = makeTruePredicate
+                        , substitution = Substitution.unsafeWrap
+                            [ (Mock.xMap, mkMap [])
+                            ]
+                        }
+                    ]
+        actual <- matchCons
+            ( mkVar Mock.xMap
+            `cons` mkMap
+                [ (mkKey 1, mkVal 11)
+                , (mkKey 2, mkVal 12)
+                , (mkKey 3, mkVal 13)
+                ]
+            )
+            [ (1, 11)
+            , (2, 12)
+            , (3, 13)
+            ]
+        assertEqualWithExplanation "" expect actual
+    ]
+  where
+    top = Just $ OrPredicate.fromPredicate Predicate.topPredicate
+    substitution subst = Just $ MultiOr.make
+        [ Conditional
+            { term = ()
+            , predicate = makeTruePredicate
+            , substitution =
+                Substitution.unsafeWrap
+                    ((fmap . fmap) mkVal subst)
+            }
+        ]
+    mkKey k =
+        (mkBuiltin . Domain.BuiltinInt)
+            Domain.InternalInt
+                { builtinIntSort  = Mock.intSort
+                , builtinIntValue = k
+                }
+    mkVal = Int.asInternal Mock.intSort
+    mkMap = Map.asInternal Mock.metadataTools Mock.mapSort . Map.fromList
+    mapWithKey = Bifunctor.bimap mkKey
+    matchMap = matchDefinition `on` mkMap
+    matchConcrete =
+        matchMap `on` fmap (mapWithKey mkVal)
+    matchVariable var val =
+            matchMap
+            (mapWithKey (either mkVar mkVal) <$> var)
+            (mapWithKey mkVal <$> val)
+    cons = Mock.concatMap
```

Comment:
Just curious: Why cons instead of concat/concat'/whatever?

Reply:
Forgot to rename it. I started with it as cons, decided concat is better but forgot to rename. Fixed.